Imageview/app/src/main/java/com/mycompany/imageview/MySQLiteHelper.java

```diff
@@ -0,0 +1,180 @@
+    }
+
+    @Override
+    public void onUpgrade(SQLiteDatabase db, int oldVersion, int newVersion) {
+        // Drop older books table if existed
+        db.execSQL("DROP TABLE IF EXISTS "+ TABLE_CONTACTS);
+
+        // create fresh books table
+        this.onCreate(db);
+    }
+
+
+
+
+    public void add_image(Image_DB image){
+        //for logging
+        Log.d("add_image", image.toString());
+
+        // 1. get reference to writable DB
+        SQLiteDatabase db = this.getWritableDatabase();
+
+        // 2. create ContentValues to add key "column"/value
+        ContentValues values = new ContentValues();
+        values.put(KEY_keyword, image.getkeyword()); // get title
+        values.put(KEY_link, image.getLink()); // get author
+
+        // 3. insert
+        db.insert(TABLE_CONTACTS, // table
+                null, //nullColumnHack
+                values); // key/value -> keys = column names/ values = column values
+
+        // 4. close
+        db.close();
+    }
+    public Image_DB get_image(int id){
+
+        // 1. get reference to readable DB
+        SQLiteDatabase db = this.getReadableDatabase();
+
+        // 2. build query
+        Cursor cursor =
+                db.query(TABLE_CONTACTS, // a. table
+                        COLUMNS, // b. column names
+                        " id = ?", // c. selections
+                        new String[] { String.valueOf(id) }, // d. selections args
+                        null, // e. group by
+                        null, // f. having
+                        null, // g. order by
+                        null); // h. limit
+
+        // 3. if we got results get the first one
+        if (cursor != null)
+            cursor.moveToFirst();
+
+        // 4. build book object
+        Image_DB image = new Image_DB();
+        image.set_id(Integer.parseInt(cursor.getString(0)));
+        image.set_keyword(cursor.getString(1));
+        image.set_link(cursor.getString(2));
+
+        //log
+        Log.d("getBook(" + id + ")", image.toString());
+
+        // 5. return book
+        return image;
+    }
+    public List<Image_DB> get_all_images() {
+        List<Image_DB> images = new LinkedList<Image_DB>();
+
+        // 1. get reference to readable DB
+        SQLiteDatabase db = this.getReadableDatabase();
+
+        // 2. build query
+        Cursor cursor =
+                db.query(TABLE_CONTACTS, // a. table
+                        COLUMNS, // b. column names
+                        null, // c. selections
+                        null, // d. selections args
+                        null, // e. group by
+                        null, // f. having
+                        null, // g. order by
+                        null); // h. limit
+
+        // 3. if we got results get the first one
+        if (cursor != null)
+            cursor.moveToFirst();
+
+
+        Image_DB image = null;
+        if (cursor.moveToFirst()) {
+            do {
```

Comment:
You can just use just a while loop here as well, since cursor.getNext() returns a boolean to let you know if the cursor has reached the end.